### DISCUSSION TITLE
ResizeObserver polyfill for certain environments

### DISCUSSION THREAD

**Discussion by RubenStoesser** [TanStack/virtual#367] [Q&A] [answered]
In [v3.0.0-beta.15](https://github.com/TanStack/virtual/releases/tag/v3.0.0-beta.15) a PR was merged introducing the [usage of ResizeObserver](https://github.com/TanStack/virtual/blob/beta/packages/virtual-core/src/index.ts#L84).

This works fine in modern browsers. However, it is causing some issues in test environments (e.g. Jest/JSDom) as `ResizeObserver` is not implemented there, so a polyfill is needed to get tests running. Without a polyfill every test will fail with:
```ts
ReferenceError: ResizeObserver is not defined
```

Your internal tests are passing because the default `observeElementRect` function which calls `new ResizeObserver` is overwritten in the `useVirtualizer` call [within the test file](https://github.com/TanStack/virtual/blob/beta/packages/react-virtual/__tests__/index.test.tsx#L36).

My question is, do you see polyfilling of ResizeObserver functionality as a responsibility of package consumers or would you consider adding a polyfill directly within the package?

Cheers & thank you for the work on this great library!

**Comment by tannerlinsley** [comment, marked as answer]
Pretty sure that falls on the application env, not ours.

Tanner Linsley
